From 606c4aa3ae9630f198377d5bfddd6ede3cffc2f0 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Wed, 7 Mar 2018 23:05:36 +0530
Subject: [PATCH] build: Fix enumtypes on MinGW inside MSYS Windows

The glib-mkenums command starts with:

--fhead "/* Generated by glib-mkenums. [...]"

The MSYS shell thinks the / is a path, and expands it to the / inside
MSYS, which is C:/MinGW/msys/1.0 and then proceeds to convert the \n
to /n and so on, which totally mangles the file header and leads to an
unusable soup-enum-types.h

Fix this by adding a space before that, so that it becomes:

--fhead " /* Generated by glib-mkenums. [...]"

Which avoids the translation. This broke because glib-mkenums is now
a Python script instead of a Perl script. The aforementioned
translation is only done for arguments passed to non-msys programs
(like python) and not for msys ones (like perl and awk).

---
 Makefile.glib | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.glib b/Makefile.glib
index 97e94b0..bd6c6c6 100644
--- a/Makefile.glib
+++ b/Makefile.glib
@@ -202,7 +202,7 @@ $(if $(_glib_enum_types_sources),,$(error Need to define $(_glib_enum_types_sour
 
 $(1).h.stamp: $(_glib_enum_types_h_sources) Makefile
 	$$(_GLIB_V_GEN) $$(GLIB_MKENUMS) \
-		--fhead "/* Generated by glib-mkenums. Do not edit */\n\n#ifndef $(_glib_enum_types_guard)\n#define $(_glib_enum_types_guard)\n\n" \
+		--fhead " /* Generated by glib-mkenums. Do not edit */\n\n#ifndef $(_glib_enum_types_guard)\n#define $(_glib_enum_types_guard)\n\n" \
 		$$(GLIB_MKENUMS_H_FLAGS) \
 		$$($(_glib_enum_types_prefix)_MKENUMS_H_FLAGS) \
 		--fhead "#include <glib-object.h>\n\nG_BEGIN_DECLS\n" \
@@ -219,7 +219,7 @@ $(1).h: $(1).h.stamp
 
 $(1).c.stamp: $(_glib_enum_types_h_sources) Makefile
 	$$(_GLIB_V_GEN) $$(GLIB_MKENUMS) \
-		--fhead "/* Generated by glib-mkenums. Do not edit */\n\n#ifdef HAVE_CONFIG_H\n#include <config.h>\n#endif\n\n#include \"$(notdir $(1)).h\"\n\n" \
+		--fhead " /* Generated by glib-mkenums. Do not edit */\n\n#ifdef HAVE_CONFIG_H\n#include <config.h>\n#endif\n\n#include \"$(notdir $(1)).h\"\n\n" \
 		$$(GLIB_MKENUMS_C_FLAGS) \
 		$$($(_glib_enum_types_prefix)_MKENUMS_C_FLAGS) \
 		--fhead "$$(foreach f,$$(filter-out Makefile,$$(^F)),\n#include \"$$(f)\")\n\n" \
-- 
2.12.2.windows.2

